### PR TITLE
Handle User Bluetooth Disconnection Request

### PR DESCRIPTION
**Pull request by cvcore**
This PR addresses the issue that the keyboard will keep reconnecting itself to the computer, even if the user disconnects it manually from the bluetooth menu.

Related issue: https://github.com/KinesisCorporation/Adv360-Pro-ZMK/issues/172#issuecomment-1646550785

The previous behavior of zmk in the adv360-z3.2 branch will start open BLE advertising, when there exists no connection for the current profile, or if the current profile is open. This behavior will cause the keyboard to broadcast itself as connectable immediately after user BLE disconnection, causing the above mentioned behavior.

With the proposed change, the keyboard will no longer advertise itself as connectable after BLE disconnection. But wait until the user presses a key on the keyboard to retrigger the BLE advertising.

@ReFil please review and give suggestions.

### CODE REVIEW

File: app/src/ble.c [hunk 1]
```diff
@@ -48,7 +50,7 @@ static uint8_t passkey_digit = 0;
 #endif /* IS_ENABLED(CONFIG_ZMK_BLE_PASSKEY_ENTRY) */
 
 #if IS_ENABLED(CONFIG_ZMK_SPLIT_ROLE_CENTRAL)
-#define PROFILE_COUNT (CONFIG_BT_MAX_PAIRED - 1)
+#define PROFILE_COUNT (CONFIG_BT_MAX_PAIRED - 1) // should it be ZMK_BLE_PROFILE_COUNT?
```
Comment (ReFil) [inline app/src/ble.c]: They are functionally the same, one pairing slot is reserved for the right half

File: app/src/ble.c [hunk 2]
```diff
@@ -63,12 +65,14 @@ enum advertising_type {
...
-                    BT_GAP_ADV_FAST_INT_MAX_2, NULL)
+                    BT_GAP_ADV_FAST_INT_MAX_2, NULL) // 100 - 150 ms advertising interval
 
+/* How are static variables stored across boots? */
 static struct zmk_ble_profile profiles[ZMK_BLE_PROFILE_COUNT];
 static uint8_t active_profile;
+bool active_profile_seeking_connection = true;
 
-#define DEVICE_NAME CONFIG_BT_DEVICE_NAME
+#define DEVICE_NAME CONFIG_BT_DEVICE_NAME // where is this defined?
```
Comment (ReFil) [inline app/src/ble.c]: anything prefixed with CONFIG_ gets set in one of the KConfig files, so adv360_*side*_defconfig as well as a few other files in the adv360 directory, a lot of parameters are set at compile time by the kconfig file in the root of app as well

File: app/src/ble.c [hunk 3]
```diff
@@ -141,6 +146,7 @@ bool zmk_ble_active_profile_is_connected() {
         return err;                                                                                \
     }
 
+/* Start direct advertising to peer if no connection exists for active profile */
```
Comment (ReFil) [inline app/src/ble.c]: I believe direct advertising to hosts is currently not a thing because apple do not permit it to any apple device

**Comment by ReFil** [comment]
It all looks good, but I'd like to se it behind a Kconfig option so as not to introduce a potentially confusing change to regulat users

**Comment by cvcore** [comment]
@ReFil Thank you for the review and clarifying open questions! Sure, adding a default off switch is definitely better idea before it's introduced to everyone as a default.
I will make the adjustments in subsequent commits and let you know when it's ready for merging.

File: app/src/ble.c (hunk 1 above)
Comment (cvcore) [reply app/src/ble.c]: Yes, but I can't find where `PROFILE_COUNT` was used anywhere in the code. Seems everywhere we are referring to `ZMK_BLE_PROFILE_COUNT` which made me a bit confused.

**Comment by cvcore** [comment]
@ReFil Added the config with switch `CONFIG_ZMK_HANDLE_BLE_DISCONNECTION`, please have another review. Thanks!

File: app/src/ble.c (hunk 1 above)
Comment (ReFil) [reply app/src/ble.c]: The number of profiles configured in ZMK is not the same as the technical limitation on the number of profiles. The PROFILE_COUNT value is defined as compile time based on the max pairing slots configured, you can have 8 pairing slots and then use ZMK_BLE_PROFILE_COUNT to limit the number of actual profiles

**Comment by ReFil** [comment]
You need to add the config option in the root KConfig otherwise the build system doesn't know whaat to do with it.

**Comment by cvcore** [comment]
> You need to add the config option in the root KConfig otherwise the build system doesn't know whaat to do with it.

Good catch! Fixed 👍

**Comment by ReFil** [comment]
Can you remove the extra comments you've added in some of the source files to streamline the changes please?

**Comment by cvcore** [comment]
Done :) lmk if any further change is needed.

**Comment by ReFil** [comment]
Please add back the line ending in app/KConfig that got removed

**Comment by cvcore** [comment]
> Please add back the line ending in app/KConfig that got removed

Hmm interesting. I've reverted the change assuming you would prefer a cleaner git history. Apart from that is there a reason for having it there? Since most modern editors will remove trailing whitespaces by default. Just asking to avoid future confusions.

Actually I didn't notice the edit in the beginning since it's done automatically from VSCode, and upstream [Kconfig](https://github.com/zmkfirmware/zmk/blob/main/app/Kconfig) also doesn't have the trailing line.

**Comment by ReFil** [comment]
Thank you, I didn't know about the upstream change. I think the cleaner change in this PR the better to try and minimise the number of merge conflicts I'll have to deal with when upstream is merged again. I'll test this PR soon and if all is good I'll merge it in

**Comment by cvcore** [comment]
FYI @ReFil : now updated the code base to `adv360-z3.2-beta` and tested on my hardware.

**Comment by cvcore** [comment]
Closed for #9

**Pull request closed without merging**